app/src/main/java/org/mascota/ui/view/diary/adpter/SelectProfileAdapter.kt

```diff
@@ -0,0 +1,96 @@
+package org.mascota.ui.view.diary.adpter
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.RecyclerView
+import org.mascota.R
+import org.mascota.databinding.ItemProfileBinding
+import org.mascota.ui.view.diary.data.ProfileList
+import kotlin.coroutines.coroutineContext
+
+class SelectProfileAdapter : RecyclerView.Adapter<SelectProfileAdapter.SelectProfileViewHolder>() {
+
+
+    var profileList = mutableListOf<ProfileList>()
+    private lateinit var itemClickListener : OnItemClickListener
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SelectProfileViewHolder {
+        //뷰홀더
+        //뷰에 대한 정보(layou,data)담고 있는 상자이자
+        //이를 넣어주는 역할
+        //databinding 객체 만들어서 item_profile.xml에 뷰로 만들어줄 데이터를 담아줘야함
+        //databidding해서 이곳을 객체로 넘겨줘야함
+
+
+        val binding : ItemProfileBinding = DataBindingUtil.inflate(
+            LayoutInflater.from(parent.context),
+            R.layout.item_profile, parent, false)
+
+        return SelectProfileViewHolder(binding)
+    }
+
+
+
+    inner class SelectProfileViewHolder(
+
+        private val viewBiding : ItemProfileBinding
+
+        ) : RecyclerView.ViewHolder(
+        viewBiding.root
+    ){
+        fun onBind(data : ProfileList, position: Int){
+            viewBiding.animal = data
+            viewBiding.tvName.setText(data.tv_name)
+            viewBiding.ivProfile.setBackgroundResource(data.img_profile)
+            viewBiding.listView.setOnClickListener {
+                // 색깔 바뀌게 했는데
+                val item = profileList.get(position).tv_name
+                itemClickListener.onClick(it, position, item)
+
+                viewBiding.tvName.isSelected = !it.isSelected
+
+                //Log.d("이름","아이템:")
+
+            }
```

Comment:
이렇게 중복되서 많이 특정 변수나 함수가 많이 호출될 경우 scope함수 활용해서 간결하게 표현하면 좋을것같아요!
```suggestion
            viewBiding.apply { 
            animal = data
            tvName.setText(data.tv_name)
            ivProfile.setBackgroundResource(data.img_profile)
            listView.setOnClickListener {
                // 색깔 바뀌게 했는데
                val item = profileList.get(position).tv_name
                itemClickListener.onClick(it, position, item)

                tvName.isSelected = !it.isSelected

                //Log.d("이름","아이템:")

            }
            }
```